runtime/lua/vim/treesitter/highlighter.lua

```diff
@@ -370,7 +407,9 @@ local function on_line_impl(self, buf, line, on_spell, on_conceal)
             conceal = conceal,
             spell = spell,
             url = url,
-          })
+          }
+          local mark = { begin_line = start_row, begin_col = start_col, opts = opts }
```

Comment:
start/end are the normal convention in the codebase

```suggestion
          local mark = { start_line = start_row, start_col = start_col, opts = opts }
```